hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/LruBlockCache.java

```diff
@@ -947,7 +1019,47 @@ public void run() {
         }
         LruBlockCache cache = this.cache.get();
         if (cache == null) break;
-        cache.evict();
+        bytesFreed = cache.evict();
+        long stopTime = System.currentTimeMillis();
+        // If heavy cleaning BlockCache control.
+        // It helps avoid put too many blocks into BlockCache
+        // when evict() works very active.
+        if (stopTime - startTime <= 1000 * 10 - 1) {
+          mbFreedSum += bytesFreed/1024/1024;
+        } else {
+          freedDataOverheadPercent =
```

Comment:
Whats the -100 part in the ending there? I think what you are computing here is what % of evictionSizeMbLimit that is filled?

Reply:
I thought it is better shows which overhead we have. When we use -100 it looks clearly for me because can show values ​​below zero. But it is possible to change, if you think of this better for understanding.

Reply:
>if (mbFreedSum > heavyEvictionMbSizeLimit)

ok, will fix it soon